django_mongodb/query.py

```diff
@@ -115,7 +115,7 @@ def join(self, compiler, connection):
                 # The pipeline variables to be matched in the pipeline's
                 # expression.
                 "let": {
-                    f"{parent_template}{i}": parent_field
+                    f"{parent_template}{i}": {"$toString": parent_field}
```

Comment:
I'm not sure that doing toString unconditionally is best. I think Django's SQL joining logic examines the the target field and only casts as necessary. Did you investigate this?

Reply:
Not much to be honest, I just asume that the field can be casted as string, but I will do a research of this.

Reply:
I found that I have to refactor Cast in order to support it. There is a tiny difference in our flow and postgres (or other sql) flow. But It is doable.

Reply:
Never mind, We can't cast to objectId but we can meet in the middle casting to string if the type are different.